dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/plugins/internal/AngleBracket.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.dokuwiki.syntax.plugins.internal;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.contrib.dokuwiki.syntax.DokuWikiSyntaxParserHelper;
+import org.xwiki.contrib.dokuwiki.syntax.plugins.DokuWikiPlugin;
+import org.xwiki.contrib.dokuwiki.syntax.plugins.internal.angleBrackets.DokuWikiAngleBracketPlugin;
+import org.xwiki.rendering.listener.Listener;
+
+/**
+ * DokuWiki Angle plugin parser.
+ *
+ * @version $Id: $
+ * @since 1.2
+ */
+@Component
+@Named("angleBracket")
+@Singleton
+public class AngleBracket implements DokuWikiPlugin
+{
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private DokuWikiSyntaxParserHelper helper;
+
+    @Inject
+    @Named("context")
+    private Provider<ComponentManager> componentManagerProvider;
+
+    private List<DokuWikiAngleBracketPlugin> componentList;
```

Comment:
Since this component is a singleton everything you put in a field will be shared by all thread currently doing sokuwiki parser which will generate a real mess so you should either kip this list as an local variable or make the component per lookup (see https://extensions.xwiki.org/xwiki/bin/view/Extension/Component+Module#HRegisteringandUsingaProvider).